core/domain/services/assets/CoreAssetManager.php

```diff
@@ -0,0 +1,357 @@
+
+    const JS_HANDLE_EE_WP_PLUGINS_PAGE = 'ee-wp-plugins-page';
+
+    // EE CSS assets handles
+    const CSS_HANDLE_EE_DEFAULT = 'espresso_default';
+
+    const CSS_HANDLE_EE_CUSTOM  = 'espresso_custom_css';
+
+    /**
+     * @var EE_Currency_Config $currency_config
+     */
+    protected $currency_config;
+
+    /**
+     * @var EE_Template_Config $template_config
+     */
+    protected $template_config;
+
+
+    /**
+     * CoreAssetRegister constructor.
+     *
+     * @param AssetCollection    $assets
+     * @param EE_Currency_Config $currency_config
+     * @param EE_Template_Config $template_config
+     * @param DomainInterface    $domain
+     * @param Registry           $registry
+     */
+    public function __construct(
+        AssetCollection $assets,
+        EE_Currency_Config $currency_config,
+        EE_Template_Config $template_config,
+        DomainInterface $domain,
+        Registry $registry
+    ) {
+        $this->currency_config = $currency_config;
+        $this->template_config = $template_config;
+        parent::__construct($domain, $assets, $registry);
+    }
+
+
+    /**
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    public function addAssets()
+    {
+        $this->addManifestFile();
+        $this->addJavascriptFiles();
+        $this->addStylesheetFiles();
+    }
+
+
+    /**
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    public function addJavascriptFiles()
+    {
+        $this->loadCoreJs();
+        $this->loadJqueryValidate();
+        $this->loadAccountingJs();
+        add_action(
+            'AHEE__EventEspresso_core_services_assets_Registry__registerScripts__before_script',
+            array($this, 'loadQtipJs')
+        );
+        $this->registerAdminAssets();
+    }
+
+
+    /**
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    public function addStylesheetFiles()
+    {
+        $this->loadCoreCss();
+    }
+
+
+    /**
+     * core default javascript
+     *
+     * @since $VID:$
+     * @throws DuplicateCollectionIdentifierException
+     * @throws InvalidArgumentException
+     * @throws InvalidDataTypeException
+     * @throws InvalidEntityException
+     */
+    private function loadCoreJs()
+    {
+        $this->addJavascript(
+            CoreAssetManager::JS_HANDLE_EE_MANIFEST,
+            $this->registry->getJsUrl($this->domain->assetNamespace(), 'manifest')
+        );
+
+        $this->addJavascript(
+            CoreAssetManager::JS_HANDLE_EE_JS_CORE,
+            $this->registry->getJsUrl($this->domain->assetNamespace(), 'eejs'),
```

Comment:
This comment applies to everywhere  you invoke `$this->registry->getJsUrl`:  the second argument is the script handle that the url is being retrieved for, so you can use the constants you created for that.  So for instance, in this specific case  you'd end up with something like (assuming you use `self` like I mentioned in another comment in my review):

```js
$this->addJavascript(
    self::JS_HANDLE_EE_JS_CORE,
    $this->registry->getJsUrl($this->domain->assetNamespace(), self::JS_HANDLE_EE_JS_CORE),
    array(self::JS_HANDLE_EE_MANIFEST)
)-> setRequiresTranslation()
    -> setHasLocalizedData()
```

Reply:
> the second argument is the script handle that the url is being retrieved for

No the second argument is the "chunk" name. 
Here's the actual method signature:

```php
public function getJsUrl($namespace, $chunk_name)
```

Here is the original code that the above snippet replaces:

```php
wp_register_script( 
    'eejs-core', 
    $this->getJsUrl(self::ASSET_NAMESPACE, 'eejs'), 
    array('ee-manifest'), 
    null, 
    true 
); 
```

As you can see, the asset handle is  `'eejs-core'` but the second parameter passed to `getJsUrl()` is `'eejs'`.

That said, if you want to refactor how the manifest files are built so that the chunk name is the same as the asset handle, then that would certainly make things easier and less error prone since we would only need to reference that one identifier. 

I would enthusiastically support your changing the manifest files to work this way